### ISSUE TITLE
Snyk: Regular Expression Denial of Service (ReDoS)

### ISSUE THREAD

**Issue by velsonjr** [fomantic/Fomantic-UI#2891]
When our application was run through Snyk recently, We found Regular Expression Denial of Service vulnerability.

Found in: fomantic/src/definitions/behaviors/api.js (line : 359)
Line would be `requiredVariables = url.match(settings.regExp.required);`
Regex would be `/\{\$*[A-z0-9]+\}/g`

Found in: fomantic/src/definitions/behaviors/api.js (line : 360)
Line would be `optionalVariables = url.match(settings.regExp.optional);`
Regex would be `/\{\/\$*[A-z0-9]+\}/g`

Description:Unsanitized user input from data from a remote resource flows into match, where it is used to build a regular expression. This may result in a Regularexpression Denial of Service attack (reDOS).

The version would be "fomantic-ui": "^2.9.2"

**Comment by lubber-de** [comment]
Mmh, any advise how to change the default regexp, so snyk does not complain about?
Because i just used https://devina.io/redos-checker to check both regexes (in 2.9.2 and in current develop branch) and both seem to be safe

At the moment i have no idea how we should otherwise sanitise the url string before using .match() onto it.

**Comment by velsonjr** [comment]
I had to basically change the following lines of code , 

```
// Adding the regExp to the list of internal aliases on line 57
regex           = settings.regExp
```
and use them like this

```
requiredVariables = url.match(regex.required);
optionalVariables = url.match(regex.optional);
```
I think the values were not getting referenced properly earlier. With this change , snyk doesnt complain about the above mentioned vulnerability.
```

**Comment by lubber-de** [comment]
Should be fixed by #2894 then
